File: src/main/java/de/tectoast/emolga/commands/soullink/StatusCommand.java
package de.tectoast.emolga.commands.soullink;

import de.tectoast.emolga.commands.Command;
import de.tectoast.emolga.commands.CommandCategory;
import de.tectoast.emolga.commands.GuildCommandEvent;
import de.tectoast.jsolf.JSONObject;

public class StatusCommand extends Command {

    public StatusCommand() {
        super("status", "Setzt den Status eines Encounters", CommandCategory.Soullink);
        setArgumentTemplate(ArgumentManagerTemplate.builder()
                .add("location", "Location", "Die Location", ArgumentManagerTemplate.Text.any())
                .add("status", "Status", "Der Status", ArgumentManagerTemplate.Text.of(SubCommand.of("Team"), SubCommand.of("Box"), SubCommand.of("RIP")))
                .setExample("/status Route 1 RIP")
                .build());
        slash(true, 695943416789598208L);
    }

    @Override
    public void process(GuildCommandEvent e) throws Exception {
        JSONObject soullink = getEmolgaJSON().getJSONObject("soullink");
        ArgumentManager args = e.getArguments();
        String location = eachWordUpperCase(args.getText("location"));
        if (!soullink.getStringList("order").contains(location)) {
            e.reply("Diese Location ist derzeit nicht im System!");
            return;
        }
        soullink.getJSONObject("mons").getJSONObject(location).put("status", args.getText("status"));
        e.reply("\uD83D\uDC4D");
        saveEmolgaJSON();
        updateSoullink();
    }
}

File: src/main/kotlin/de/tectoast/emolga/commands/soullink/StatusCommand.kt
package de.tectoast.emolga.commands.soullink

import de.tectoast.emolga.commands.Command
import de.tectoast.emolga.commands.CommandCategory
import de.tectoast.emolga.commands.GuildCommandEvent
import de.tectoast.emolga.commands.saveEmolgaJSON
import de.tectoast.emolga.utils.json.Emolga

class StatusCommand : Command("status", "Setzt den Status eines Encounters", CommandCategory.Soullink) {
    init {
        argumentTemplate = ArgumentManagerTemplate.builder()
            .add("location", "Location", "Die Location", ArgumentManagerTemplate.Text.any())
            .add(
                "status",
                "Status",
                "Der Status",
                ArgumentManagerTemplate.Text.of(SubCommand.of("Team"), SubCommand.of("Box"), SubCommand.of("RIP"))
            )
            .setExample("/status Route 1 RIP")
            .build()
        slash(true, 695943416789598208L)
    }

    override suspend fun process(e: GuildCommandEvent) {
        val soullink = Emolga.get.soullink
        val args = e.arguments
        val location = eachWordUpperCase(args.getText("location"))
        if (location !in soullink.order) {
            e.reply("Diese Location ist derzeit nicht im System!")
            return
        }
        soullink.mons[location]!!["status"] = args.getText("status")
        e.reply("\uD83D\uDC4D")
        saveEmolgaJSON()
        updateSoullink()
    }
}